### PR TITLE
refactor: separate endpoint

### PR DESCRIPTION
**Pull request by Rae-Kwon**
EDIT: This separates the endpoint so the base finnhub API URL is in the StockListing page and only the secret key and token query are placed in the .env file (key omitted from .env.example file). 

To get stock symbol info for example you would do `${finnhubURL}/stock/symbol?exchange=US${process.env.REACT_APP_STOCK_TOKEN}`

To get stock quote price info for example you would do`${finnhubURL}/quote?symbol=AAPL${process.env.REACT_APP_STOCK_TOKEN}`

Note that the & before the stock token is omitted as it is baked into REACT_APP_STOCK_TOKEN env variable.

 ̶T̶h̶i̶s̶ ̶s̶e̶p̶a̶r̶a̶t̶e̶s̶ ̶t̶h̶e̶ ̶e̶n̶d̶p̶o̶i̶n̶t̶ ̶s̶o̶ ̶t̶h̶e̶ ̶b̶a̶s̶e̶ ̶R̶E̶A̶C̶T̶_̶A̶P̶P̶_̶S̶T̶O̶C̶K̶_̶U̶R̶L̶ ̶i̶s̶ ̶j̶u̶s̶t̶ ̶h̶t̶t̶p̶s̶:̶/̶/̶f̶i̶n̶n̶h̶u̶b̶.̶i̶o̶/̶a̶p̶i̶/̶v̶1̶/̶s̶t̶o̶c̶k̶.̶ ̶A̶l̶s̶o̶ ̶a̶d̶d̶e̶d̶ ̶e̶n̶d̶p̶o̶i̶n̶t̶ ̶t̶o̶ ̶g̶a̶t̶h̶e̶r̶ ̶s̶t̶o̶c̶k̶ ̶p̶r̶i̶c̶e̶s̶.̶ ̶I̶t̶ ̶i̶s̶ ̶s̶e̶p̶a̶r̶a̶t̶e̶d̶ ̶s̶o̶ ̶y̶o̶u̶ ̶w̶o̶u̶l̶d̶ ̶h̶a̶v̶e̶ ̶t̶o̶ ̶c̶a̶l̶l̶ ̶f̶o̶r̶ ̶e̶x̶a̶m̶p̶l̶e̶ ̶i̶f̶ ̶y̶o̶u̶ ̶w̶a̶n̶t̶e̶d̶ ̶A̶p̶p̶l̶e̶'̶s̶ ̶s̶t̶o̶c̶k̶ ̶p̶r̶i̶c̶e̶:̶
̶
̶`̶$̶{̶p̶r̶o̶c̶e̶s̶s̶.̶e̶n̶v̶.̶R̶E̶A̶C̶T̶_̶A̶P̶P̶_̶S̶T̶O̶C̶K̶_̶P̶R̶I̶C̶E̶}̶=̶A̶A̶P̶L̶$̶{̶p̶r̶o̶c̶e̶s̶s̶.̶e̶n̶v̶.̶R̶E̶A̶C̶T̶_̶A̶P̶P̶_̶U̶R̶L̶_̶T̶O̶K̶E̶N̶}̶`̶

### CODE REVIEW

File: src/pages/StockListing/StockListing.tsx [hunk 1]
```diff
@@ -16,14 +16,16 @@ export type Stock = {
...
+
 //Todo: Add proper pagination
 const StockListing = (): JSX.Element => {
   const [data, setData] = useState<Stock[] | []>([]);
   const howManyToDisplay: number = 10;
 
   useEffect(() => {
     const fetchData = async () => {
-      const rawData = await fetch(process.env.REACT_APP_STOCK_URL!);
+      const rawData = await fetch(`${finnhubURL}/stock/symbol?exchange=US${process.env.REACT_APP_STOCK_TOKEN}`!);
```
Comment (JessieFrance) [inline src/pages/StockListing/StockListing.tsx]: We could even eliminate the separate token variable in `.env` and just have something like this:
```const rawData = await fetch(`${finnhubURL}/stock/symbol?exchange=US&token=${process.env.REACT_APP_STOCK_KEY}`!);```

**Review by JessieFrance — APPROVED** [review]
I just tested this and it works. The code refactor also looks good. I added a comment on how we might reduce the number of `.env` variables, but it's not really critical. As a side note (that isn't directly related to this PR), I think our group should move the API key to the backend (and just forward the same info to the frontend from there). In any case, I vote to merge this PR. Thanks for the awesome work Richard!

File: src/pages/StockListing/StockListing.tsx (hunk 1 above)
Comment (Rae-Kwon) [reply src/pages/StockListing/StockListing.tsx]: Thanks for the suggestion!! I've gone and removed the token env var and added the token query to the finnhub API call

**Review by ArturoGrullon — COMMENTED** [review]
The env. variable wasn't working?

**Comment by ArturoGrullon** [comment]
> The env. variable wasn't working?

Forget it, now I see the comments

**Review by ArturoGrullon — APPROVED** [review]

**Pull request merged**